src/getMigrateSavedFilters.ts

```diff
@@ -37,31 +37,44 @@ export const getMigrateSavedFilters =
     },
   ) =>
   <FromFilterState, ToFilterState>(
+    deserialize: (state: any) => FromFilterState,
     callback: MigrateFilterCallback<FromFilterState, ToFilterState>,
+    serialize: (state: ToFilterState) => any,
   ): void => {
-    const filtersContent =
-      contentServer.children?.ui.children?.filters.children?.content.children;
-    const filtersStructure =
-      contentServer.children?.ui.children?.filters.children?.structure!;
-    const filesAncestry = _getFilesAncestry(filtersStructure);
+    const { content, structure } =
+      contentServer.children?.ui.children?.filters.children ?? {};
 
-    const migrateFilter = produce(callback);
+    if (!content?.children || !structure?.children) {
+      return;
+    }
+
+    const filesAncestry = _getFilesAncestry(structure);
 
-    for (const fileId in filtersContent) {
-      let migratedFilter;
-      const { entry } = filtersContent[fileId];
+    for (const fileId in content.children) {
+      const { entry } = content.children[fileId];
       const filter = JSON.parse(entry.content);
 
       const folderName = filesAncestry[fileId].map(({ name }) => name);
       const folderId = filesAncestry[fileId].map(({ id }) => id);
-      const metadata = _getMetaData(filtersStructure, folderId, fileId);
+      const metadata = _getMetaData(structure, folderId, fileId);
 
       try {
-        migratedFilter = migrateFilter(filter, { dataModels });
-        // The filter was fully migrated.
+        const deserializedFilter = deserialize(filter);
+        const deserializedMigratedFilter = produce(
+          deserializedFilter,
+          (draft) => callback(draft as FromFilterState, { dataModels }),
```

Comment:
Why do we need to cast here? If `deserialize` returns `FromFilterState`

Reply:
`callback` expects a `FromFilterState`, but `TypeScript` complains that it receives a `Draft<FromFilterState>`. I think `immer` imposes that to warn against mutability, but here we know what we're doing so it's not a problem.
We could probably type `callback` as a `Recipe` instead, but that would make it harder to read for someone just wanting to write a script, and couple our types to `produce`.
So I prefer to cast and keep the simple "mutate the input" contract of `migrateXyz`.

Reply:
Yeah, I agree with you, casting sounds better.
I was gonna ask you to add a comment above to explain why we need to do it, but actually I'm not sure we need to. It makes sense that `draft` needs to be a `FromFilterState` at this point. So feel free to add one if you feel the need, but otherwise I'm happy to keep this as it is.